src/main/java/com/sysdig/jenkins/plugins/sysdig/scanner/InlineScannerRemoteExecutor.java

```diff
@@ -84,7 +87,22 @@ public String call() throws InterruptedException, AbortException {
       }
     }
 
-    ContainerRunner containerRunner = containerRunnerFactory.getContainerRunner(logger, envVars);
+    List<String> bindMounts = new ArrayList<>();
+    String dockerVolumeInContainer = null;
+
+    // see https://github.com/jenkinsci/sysdig-secure-plugin/pull/55 discussion
+    if (envVars.containsKey("DOCKER_HOST")) {
+      String candidateVolumeHostPath = envVars.get("DOCKER_HOST");
+      if (Util.isValidLocalPath(candidateVolumeHostPath)) {
```

Comment:
Do you think we should also check if the path is not only valid, but it also exists? Using the File class and f.exists() (see https://github.com/jenkinsci/sysdig-secure-plugin/blob/b0da2ce0a7f9d256f6ca3eab676d47fd01fdb1b4/src/main/java/com/sysdig/jenkins/plugins/sysdig/scanner/InlineScannerRemoteExecutor.java#L82)

If we see that the file does not even exist, it would be better to give a clear error at this point and abort, instead of letting the container run with an invalid bind mount, as probably the error will be harder to troubleshoot.

Reply:
Useless warning (pardon me): this would be an improvement but it's not 100% failsafe: if the file is a symlink it would be evaluated as existent, but docker wouldn't be able to mount it: https://stackoverflow.com/questions/38485607/mount-host-directory-with-a-symbolic-link-inside-in-docker-container

I don't think it's worth to spend time to check for this corner case (symlinks are the spawn of the evil), I only wanted to share it!

Reply:
In the end I have added a check based on `isFile()`, I am not sure how it behaves for symbolic links, but it should work if a path is given as env variable.

Reply:
My only remaining concern is that DOCKER_HOST points to a non-existing file or a symbolic link or similar, it will fail the "validFile" check and don't set the bind mount, but set the DOCKER_HOST inside the container. So this is ok for tcp:// and others, but if the path is right (i.e. /foo/bar/docker.sock) but the agent is missing the docker.sock in that path, it will just skip setting the bind mount silently, and then fail from the inline-scanner command as it cannot open the docker.sock. It would be great if we can detect the error before launching the inline-scanner and throw a meaningful error.

But probably not worth the effort at this point, the fix is good enough :)